### ISSUE TITLE
Firefox 71 breaks the extension

### ISSUE THREAD

**Issue by P1zz4br0etch3n** [keepassxreboot/keepassxc-browser#704]
Since the update to Firefox 71 the settings page is broken and selecting credentials from input elements doesn't work any more. 

![Bildschirmfoto vom 2019-12-05 10-47-15](https://user-images.githubusercontent.com/9010623/70226490-3bb92280-1751-11ea-9508-c7b128b0b4f4.png)

## Expected Behavior
Settings can be viewed and credentials can be selected by clicking into username input fields or on the KeePassXC icon inside the input field.

## Current Behavior
Only the settings header is viewed and there is neither an icon inside input fields nor a selection is opened when one clicks inside it.

## Possible Solution
I don't know. I downgraded Firefox for now.

## Steps to Reproduce (for bugs)
1. Update Firefox to version 71
2. Try to change settings
3. Try to select passwords from input fields

## Debug info

KeePassXC - 2.5.1
KeePassXC-Browser - 1.5.3
Operating system: Fedora 31
Browser: Firefox 71

**Comment by varjolintu** [comment]
What does the page inspector / JavaScript console say after opening or reloading the Settings page?

**Comment by vanous** [comment]
FF 71, KPXC-browser basically stopped working.

```
Error loading options page: Error: undefined options.js:21:17
```

**Comment by P1zz4br0etch3n** [comment]
![Bildschirmfoto vom 2019-12-05 12-07-35](https://user-images.githubusercontent.com/9010623/70230090-e3395380-1757-11ea-9d5e-70a25e7d70de.png)

The error occurs in the first line of the try block:

`const settings = await browser.runtime.sendMessage({ action: 'load_settings' });`

If I pause on caught exceptions I get the following in jQuery:

`SyntaxError: '*,:x' is not a valid selector`
`SyntaxError: '[s!='']:x' is not a valid selector`

And this in options.js

`Error: undefined`

**Comment by varjolintu** [comment]
Thanks. Good to know. I'll try updating the jQuery. Although we already have a PR that updates the Bootstrap where jQuery is no longer needed. Still, I'm releasing a new update soon so hopefully this issue will be fixed with the minor update.

**Comment by varjolintu** [comment]
Still, I'm also using Firefox 71 but have no issues. So first I'll have to try to reproduce this problem.

**Comment by P1zz4br0etch3n** [comment]
The select from input part is similar. Tried on Google login page.

Console error: `Error: Cannot load extension settings` in `keepassxc-browser.js:828:17`

And again the error occurs in the first line of the try block:
```
const settings = await browser.runtime.sendMessage({
    action: 'load_settings'
});
```
But no jQuery errors here.

**Comment by P1zz4br0etch3n** [comment]
> Still, I'm also using Firefox 71 but have no issues. So first I'll have to try to reproduce this problem.

I can reproduce this with a clean Fedora 31 install (and dnf update) in a gnome-boxes virtual machine.

**Comment by varjolintu** [comment]
I'll install Fedora 31 to a VM and give it a try.

If anyone is willing to debug this, the background script uses `page.initSettings()` to load settings from the localStorage. Something strange might happen there.

**Comment by varjolintu** [comment]
Reproduced the problem.  It fails here: https://github.com/keepassxreboot/keepassxc-browser/blob/develop/keepassxc-browser/background/page.js#L82. So for some reason Firefox cannot write the settings to localStorage, but receives them normally.

I'm quite sure this is related to https://bugzilla.mozilla.org/show_bug.cgi?id=1592897.

The test site https://firefox-storage-test.glitch.me/ gives me "Good: Totally Working" to all four steps with Firefox 71 in macOS. With Fedora 31 half of them give "Bad: Totally Broken" results for me.

**Comment by TimmiORG** [comment]
> Reproduced the problem. It fails here: https://github.com/keepassxreboot/keepassxc-browser/blob/develop/keepassxc-browser/background/page.js#L82. So for some reason Firefox cannot write the settings to localStorage, but receives them normally.
> 
> I'm quite sure this is related to https://bugzilla.mozilla.org/show_bug.cgi?id=1592897.
> 
> The test site https://firefox-storage-test.glitch.me/ gives me "Good: Totally Working" to all four steps with Firefox 71 in macOS. With Fedora 31 half of them give "Bad: Totally Broken" results for me.

All for tests are showing good for me on Fedora 31 but still I have the issue as described here.

**Comment by GregoireW** [comment]
On Fedora31 + Firefox 71 I got the same issue. The glitch.me site give me "Good" to all steps.

The core error is: `InvalidStateError: A mutation operation was attempted on a database that did not allow mutations.`

with the stack 

```
IndexedDB.jsm:120:34
    method resource://gre/modules/IndexedDB.jsm:120
    set resource://gre/modules/ExtensionStorageIDB.jsm:237
    AsyncFunctionNext self-hosted:690
    enter resource://devtools/server/actors/utils/event-loop.js:79
    enter self-hosted:873
    _pushThreadPause resource://devtools/server/actors/thread.js:250
    _pauseAndRespond resource://devtools/server/actors/thread.js:842
    pauseAndRespond resource://devtools/server/actors/thread.js:1044
    _makeOnStep resource://devtools/server/actors/thread.js:973
    initSettings moz-extension://33246c10-7066-46a1-bded-359bc85fecab/background/page.js:80
    AsyncFunctionThrow self-hosted:695
```
(the line before last is the line from the extension)

Now, I'm not sure why, (old post talk about private browsing, but I'm not in this use case) and a little bit tired to look for a solution today

**Comment by vanous** [comment]
Same here on Debian, tests passed, issue remains.

**Comment by varjolintu** [comment]
Firefox 70 still works normally?

**Comment by vanous** [comment]
yes, just downgraded and with 70 all OK.

**Comment by varjolintu** [comment]
Also, see: https://bugzilla.redhat.com/show_bug.cgi?id=1779570

**Comment by AmazingDreams** [comment]
> If I pause on caught exceptions I get the following in jQuery:
> 
> `SyntaxError: '*,:x' is not a valid selector`
> `SyntaxError: '[s!='']:x' is not a valid selector`

This one is expected according to jQuery devs

> Don't break on caught exceptions. We caught it because we expected it. 

https://bugs.jquery.com/ticket/13881#comment:1

**Comment by diraol** [comment]
Same here (Debian bullseye/sid + Firefox 71.0 running under Wayland)

**Comment by marillat** [comment]
Seems to be a firefox bug.

I can't reproduce this bug under windows 10 firefox 71 or chromium 76 with linux

I see another extensions who doesn't work with firefox 71 
Form History Control (II) manage history open a windows with loading and never end.

**Comment by varjolintu** [comment]
According to the comments of the previous Firefox bug, this is probably the cause: https://bugzilla.mozilla.org/show_bug.cgi?id=1601707

**Comment by jovandeginste** [comment]
Fixed indeed when upgrading to [firefox-71.0-15.fc31.x86_64](https://koji.fedoraproject.org/koji/buildinfo?buildID=1420745)

**Comment by P1zz4br0etch3n** [comment]
For me too. Thanks.

**Comment by vanous** [comment]
This is good to know, thank you @jovandeginste . Will need to wait before it lands in Debian, but i am happy on 70 so i don't mind waiting.

**Comment by varjolintu** [comment]
I'm closing this issue after Debian has provided a fixed update.

**Comment by fezie** [comment]
Debian has fixed this now too:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=946547

It's already available on the mirrors

**Comment by varjolintu** [comment]
Modified the title and pinned the issue so we don't get duplicate issues all the time.

**Comment by osamak** [comment]
> Fixed indeed when upgrading to [firefox-71.0-15.fc31.x86_64](https://koji.fedoraproject.org/koji/buildinfo?buildID=1420745)

Fedora has landed the new release into Fedora 31 updates repo.
That fixed this issue! :tada:
